homework/grayscale-image/compression.cpp

```diff
@@ -0,0 +1,49 @@
+#include "compression.hpp"
+
+std::vector<std::pair<uint8_t, uint8_t>> compressGrayscale(const std::array<std::array<uint8_t, width>, height>& image_array) {
+    std::vector<std::pair<uint8_t, uint8_t>> amount_of_gray;
+    amount_of_gray.reserve(image_array.size());
```

Comment:
Here you reserve memory only for one dimension size. Notice that array is 2D so you need width * height memory to be reserved